code/code/misc/damage.cc

```diff
@@ -1133,21 +1149,22 @@ int TBeing::damageEpilog(TBeing* v, spellNumT dmg_type) {
 
     // send an achievement message to the discord webhook
     sstring achievement_msg;
-    if(isPc() && !v->isPc() && v->GetMaxLevel() >= Discord::ACHIEVEMENT_THRESHOLD) {
+    if((isPc() || (isPet(PETTYPE_PET | PETTYPE_CHARM | PETTYPE_THRALL) && (master && master->isPc()))) && !v->isPc() && v->GetMaxLevel() >= Discord::ACHIEVEMENT_THRESHOLD) {
       // player killed an notable mob, send message to our discord webhook
       // ideally we'd like to list all the group members...
-      if(grouplist == "") {
-        // if no grouplist, just use the killer's name
-        grouplist = format("%s has") % killer;
+      if (group_members == 0) {
+        // killer is solo
+        achievement_msg = format(":crossed_swords: **%s** has taken down **%s**!") % getName() % v->getName();
       } else {
```

Comment:
Several ways to make it nicer.
```cpp
if (group_members == 0) {
...
} else if (group_members == 1) {
...
} else {
...
}
```
or
```cpp
achievement_msg =
  group_members == 0
  ? format(solo)
  : group_members == 1
  ? format(duo)
  : format(group)
```

Reply:
flattened this up along the lines of the first example